crates/matrix-sdk-ui/src/sync_service.rs

```diff
@@ -90,6 +111,58 @@ impl SyncTaskSupervisor {
         Self { task, termination_sender }
     }
 
+    async fn offline_check(
+        client: &Client,
+        receiver: &mut Receiver<TerminationReport>,
+    ) -> Option<TerminationReport> {
+        info!("Entering the offline mode");
+
+        let wait_for_termination_report = async {
+            loop {
+                // Since we didn't empty the channel when entering the offline mode in fear that
+                // we might miss a report with the
+                // `TerminationOrigin::Supervisor` origin and the channel might contain stale
+                // reports from one of the sync services, in case both of them have sent a
+                // report, let's ignore all reports we receive from the sync
+                // services.
+                let report =
+                    receiver.recv().await.unwrap_or_else(TerminationReport::supervisor_error);
+
+                match report.origin {
+                    TerminationOrigin::EncryptionSync | TerminationOrigin::RoomList => (),
+                    // Since the sync service aren't running anymore, we can only receive a report
+                    // from the supervisor. It would have probably made sense to have separate
+                    // channels for reports the sync services send and the user can send using the
+                    // `SyncService::stop()` method.
```

Comment:
Should we panic in the unexpected case then? Or log an error?

Reply:
No, I think it's completely normal that this will happen.

I.e. if we can't reach the server both sync services will get an error and send a termination report.

The supervisor will only handle one of those and stop everything. The second termination report will be left in the channel.

This was fine since we would drop the channel once the supervisor was stopped, but now we don't stop the supervisor and the channel contains the stale termination report.

I think it's fine to just ignore it.